### PR TITLE
relative file path of a include statement relatives to directory of including file

### PR DESCRIPTION
**Pull request by SeaTalk**
This change is another solution for  [#114](#114 )

### CODE REVIEW

**Comment by SeaTalk** [comment]
Anyone can review this change ?

**Comment by MikaelSmith** [comment]
This looks like it could work. Did you explore my idea in https://github.com/puppetlabs/cpp-hocon/pull/114#discussion_r249910325 and find it didn't help?

**Comment by MikaelSmith** [comment]
This generally seems good. @caseywilliams interested in doing a review?

**Comment by SeaTalk** [comment]
> This looks like it could work. Did you explore my idea in [#114 (comment)](https://github.com/puppetlabs/cpp-hocon/pull/114#discussion_r249910325) and find it didn't help?

I followed your idea to find out the key issue exists the isolation of context for each parseable instance. The parseable_file of included file has not any information from including file. I didn't override function "relative_to" for sub-classes of parseable since we don't have a particular  path for "resource", like java, in c++ for now.

File: lib/tests/simple_confs/a.conf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    include file("d.conf")
+    Peter : {
+        include file("b.conf")
+        include file("sub/b.conf")
+        # passwd4 : ${new_passwd}
```
Comment (MikaelSmith) [inline lib/tests/simple_confs/a.conf]: This didn't work?

File: lib/src/parseable.cc [hunk 2]
```diff
@@ -105,17 +105,38 @@ namespace hocon {
         // if it starts with "/" then remove the "/", for consistency
```
Comment (MikaelSmith) [inline lib/src/parseable.cc]: Looks like this comment is no longer accurate.

File: lib/tests/simple_confs/a.conf (hunk 1 above)
Comment (SeaTalk) [reply lib/tests/simple_confs/a.conf]: This works。I don't remember what I was testing that I put a sharp in front of it.

File: lib/src/parseable.cc (hunk 2 above)
Comment (SeaTalk) [reply lib/src/parseable.cc]: yeah, I'll try to correct it

File: lib/inc/internal/simple_includer.hpp [hunk 3]
```diff
@@ -56,8 +56,31 @@ namespace hocon {
...
                                                    config_parse_options parse_options) const = 0;
+
+        void set_context(shared_include_context context) {
+            if (!_init_finished) {
+                _init_finished = true;
+                _context = context;
+            }
+        }
+
+        bool context_initialed() {
```
[the remark this answers is not in the document]
Comment (SeaTalk) [reply lib/inc/internal/simple_includer.hpp]: > Except for one spelling thing, I think this looks okay. The test cases look thorough, so 👍

Thanks for reviewing. I'll correct that ASAP.

**Comment by SeaTalk** [comment]
Done. @MikaelSmith @Magisus

**Comment by MikaelSmith** [comment]
Thanks so much for your work on this!

**Comment by zzhang97** [comment]
I was caught by this bug too with latest release. Happily found the fix was already in master. Thanks!

**Pull request merged**